adapters/repos/db/vector/hnsw/deserializer.go

```diff
@@ -384,13 +411,19 @@ func (c *Deserializer) ReadClearLinksAtLevel(r io.Reader, res *DeserializationRe
 		// it
 		res.Nodes[id] = &vertex{
 			id:          id,
-			connections: map[int][]uint64{},
+			connections: make([][]uint64, level+1),
 		}
 	}
 
 	if res.Nodes[id].connections == nil {
-		res.Nodes[id].connections = map[int][]uint64{}
+		res.Nodes[id].connections = make([][]uint64, level+1)
 	} else {
+		if len(res.Nodes[id].connections) <= int(level) {
+			// we need to grow the connections slice
+			newConns := make([][]uint64, level+1)
+			copy(newConns, res.Nodes[id].connections)
+			res.Nodes[id].connections = newConns
+		}
```

Comment:
This is the 5th time I'm seeing this, maybe factor it out into an extra function?

Reply:
Good point, done. With the new unit test coverage, it was very comfortable to refactor this.